pkg/v1/tkg/tkgctl/create_cluster_test.go

```diff
@@ -327,7 +420,205 @@ var _ = Describe("Unit tests for - ccluster.yaml as input file for 'tanzu cluste
+				Namespace:              "",
+				GenerateOnly:           false,
+				TkrVersion:             fakeTKRVersion,
+				SkipPrompt:             true,
+				Edition:                "tkg",
+				ClusterConfigFile:      cclassInputFileAzure,
+			}
+		})
+		It("'tanzu cluster create -f cluster_azure.yaml' use case - vsphere provider - devcc plan:, make sure legacy configurations are updated..", func() {
+			/*
+			 Process input ccluster.yaml file, this should process input ccluster.yaml file
+			 and update the environment with legacy name and values
+			 most of ccluster.yaml attributes are mapped to legacy variable for more look this - constants.CclusterToLegacyVariablesMapAzure
+			*/
+			IsInputFileClusterClassBased, err := ctl.processWorkloadClusterInputFile(&options)
+			Expect(IsInputFileClusterClassBased).Should(BeTrue())
+			Expect(err).To(BeNil())
+
+			/*
+				Here, we are preparing to test above call - ctl.processWorkloadClusterInputFile(&options)
+				to make sure input ccluster attributes values are updated with legacy variables in environment (tkgctl.TKGConfigReaderWriter())
+				Processing the input ccluster.yaml file with the existing util api, and expecting Cluster YAML object,
+				from the Cluster YAML object, map attributes path with values, and update Map variablesMap.
+			*/
+			_, cclusterObj, _ := ctl.checkIfInputFileIsClusterClassBased(options.ClusterConfigFile)
+			variablesMap := make(map[string]interface{})
+			variablesMap["metadata.name"] = cclusterObj.GetName()
+			variablesMap["metadata.namespace"] = cclusterObj.GetNamespace()
+			spec := cclusterObj.Object[constants.SPEC].(map[string]interface{})
+			err = processYamlObjectAndAddToMap(spec, constants.SPEC, variablesMap)
+			Expect(err).To(BeNil())
+
+			/*
+				Here, taking legacy variable name for every attribute in ccluter input file,
+				then check the value of legacy variable in environment, with value of the same from the input file
+				both should mathc.
+			*/
+			for key := range variablesMap {
+				if inputValue := variablesMap[key]; inputValue != nil {
+					legacyNameForClusterObjectInputVariable, ok := constants.CclusterAttributesToLegacyVariablesMapAzure[key]
+					if ok && legacyNameForClusterObjectInputVariable != "" {
+						mappedVal, _ := ctl.TKGConfigReaderWriter().Get(legacyNameForClusterObjectInputVariable)
+						Expect(fmt.Sprintf("%v", mappedVal)).To(Equal(fmt.Sprintf("%v", inputValue)))
+					}
+				}
+			}
+
+			// checking manually for some variables mapping values
+			mappedVal, _ := ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableClusterName)
+			Expect("azure-workload-cluster1").To(Equal(fmt.Sprintf("%v", mappedVal)))
+			// checking manually for some variables mapping values
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableNamespace)
+			Expect("namespace-test1").To(Equal(fmt.Sprintf("%v", mappedVal)))
+			// "spec.clusterNetwork.services.cidrBlocks": ConfigVariableServiceCIDR,
```

Comment:
these comments necessary?

Reply:
Yes but not this way, I have added proper comments.